packages/blade/src/tokens/global/motion.ts

```diff
@@ -92,6 +92,18 @@ export type Motion = Readonly<{
   easing: Easing;
 }>;
 
+type DotNotationMotionStringToken<TokenType> = {
```

Comment:
Didn't want to club them together, it's two different types and it didn't make sense having `EasingFunctionFactory | ColorContrast | string` what do you think?

Reply:
It makes sense nah? You are tokenising something here. And DotNotationString type should suffice all the cases 🤔  As long as it's not bugging out it should be fine.

Reply:
Still not too sure, there would be more use-cases where we'd need dot notation type, do we keep adding more and more recursion cases like `EasingFunctionFactory | ColorContrast | string | SomethingNew | SomethingElse`.  This would just keep bloating up and the type would make very little sense. It could also become a single point of failure. Making change in one type, we have to hope and check nothing else has broken

Reply:
Yeah this is exactly what we should do `EasingFunctionFactory | string | SomethingNew | SomethingElse `
We are tokenising the theme file right? So all the valid leaf nodes should be in the union. 

We are exporting motion also nah.
```
const bankingTheme: ThemeTokens = {
  border,
  breakpoints,
  colors,
  motion, // <-- 
  spacing,
  shadows,
  typography,
};
```

otherwise we will just end up duplicating this complex type 3 times and maintaining them separately. 

> It could also become a single point of failure. Making change in one type, we have to hope and check nothing else has broken

Currently we don't have enough divergence to duplicate and separate out this type to two different types, only adding an extra union works in both cases. As we go forward we will be adding type tests so that we don't have to hope nothing breaks.

Reply:
I'm not sure. Even right now just the color dot notation has so many permutations that its hard to know if we broke something or not. 

> all the valid leaf nodes should be in the union

That seems very scary and very difficult to maintain. If even 1 of our paths to the leaf node becomes complex (For example we decide we want to add a lot more branches to shadows before it reaches a leaf node) we would have to modify the common DotNotation type which is already being used in way too many places and could lead to breakage. 

This is also a very critical type since its consumer-facing and even 1 incorrect acceptable value would lead to an incorrect value being accepted by some component that could lead to breakage in that component. I'd rather scope the type to one path to ensure our impact radius remains minimal

@kamleshchandnani thoughts?

Reply:
umm, duplicating has its own downsides, imagine we surface a bug in DotNotation we need to fix it in all the DotNotation utils which is scary. 

@chaitanyadeorukhkar why is putting union be tricky?

```typescript
[K in keyof TokenType]: `${Extract<K, string>}.${TokenType[K] extends Record<
    string,
    string | EasingFunctionFactory | Something | SomethingElse
  >
    ? Extract<keyof TokenType[K], string | EasingFunctionFactory | Something | SomethingElse>
    : DotNotationMotionStringToken<TokenType[K]>}`;
}[keyof TokenType];
```

shouldn't the above work?

of if we want to optimise for reader then maybe we can have multiple DotNotation types one for each token type in a single file and then expose the union 

```typescript
type DotNotationColorStringToken<TokenType> = DotNotationCommonTokens<TokenType> | DotNotationMotionTokens<TokenType>
```

Reply:
+1 for composing multiple types  `DotNotationCommonTokens<TokenType> | DotNotationMotionTokens<TokenType>`

Also even if we decided to duplicate we should keep these two close together. So that changes are propagated to both at once and none of them is left. We can have a common file which holds these two types for dot tokenization. 

Plus, while I was messing around with types yesterday saw that `DotNotationMotionStringToken ` and `DotNotationColorStringToken` doesn't handle 1 level deep objects, that's why we did this: 
```ts
type DurationStringTokens = `duration.${keyof Duration}`
```



Here is a totally complete & generic dot tokenization type, this can tokenise our whole theme file: 

```ts
type DotNotationColorStringToken<TokenType> = TokenType extends Record<string, string | number>
  ? `${Extract<keyof TokenType, string>}`
  : {
      [K in keyof TokenType]: `${Extract<K, string>}.${TokenType[K] extends Record<
        string,
        string | EasingFunctionFactory | number
      >
        ? Extract<keyof TokenType[K], string | EasingFunctionFactory | number>
        : DotNotationColorStringToken<TokenType[K]>}`;
    }[keyof TokenType];
```

Reply:
Discussed & concluded that we will keep both of these types in `types.ts` with two different types for color strings & motion strings - this will ensure we don't allow any unintended types due to some error